File: LeetCode/1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        n = len(s)
        answer = s[0]
        prev_ch = s[0]
        cnt = 1

        for i in range(1,n):
            if s[i] == prev_ch:
                cnt+=1
            else:
                prev_ch = s[i]
                cnt = 1
            
            if cnt < 3:
                answer += s[i]
        
        return answer

File: LeetCode/1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        string answer;
        answer += s[0];
        int n = s.length();
        char prev_ch = s[0];
        int cnt = 1;

        for(int i = 1;i<n;i++){
            if(prev_ch == s[i])
                cnt++;
            else{
                prev_ch = s[i];
                cnt = 1;
            }

            if(cnt < 3)
                answer += s[i];
        }
        

        return answer;
    }
};